Acme.Interface.WebAPI/Controllers/v1/BooksController.cs

```diff
@@ -30,16 +31,23 @@ public BooksController(IBooksApiService booksApiService)
     /// <summary>
     /// Gets all books paged asynchronously.
     /// </summary>
-    /// <param name="pagingDto">The paged request DTO.</param>
+    /// <param name="page">The page.</param>
+    /// <param name="pageSize">Size of the page.</param>
+    /// <param name="sortColumn">The sort column.</param>
+    /// <param name="sortDirection">The sort direction.</param>
     /// <returns>
     /// Paged books.
     /// </returns>
-    [HttpPost("search")]
+    [HttpGet("search")]
     [ProducesResponseType(typeof(PagedResponseDto<BookDto>), StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<IActionResult> GetAllAsync([FromBody] PagedRequestDto pagingDto) =>
-        this.Ok(await this.booksApiService.GetAllAsync(pagingDto));
+    public async Task<IActionResult> GetAllAsync(
+        int page = 0,
+        int pageSize = 20,
+        string sortColumn = "Id",
+        SortDirection sortDirection = SortDirection.Ascending) =>
+        this.Ok(await this.booksApiService.GetAllAsync(new PagedRequestDto(page, pageSize, sortColumn, sortDirection)));
```

Comment:
I'm not sure if this is the best way of doing so, defining them like this explicitly.

Consider creating a new model in the API layer, called `PagedRequestQuery` for example, that would have these properties you defined just prefixed with the attribute `[FromQuery]`.

E.g.
```
public record PagedRequestQuery
{
    [FromQuery]
    public int Page { get; init; } = 0;
    
    [FromQuery]
    public int PageSize {get; init; } = 20;
    
    ...
}
```

This way you would keep the controller simple and pretty much the same like it was:

```
...
[HttpGet]
public async Task<IActionResult> GetAllAsync([FromQuery] pagedRequestQuery)
...
```

Reply:
Good point, we even had a PagedRequestDto from before, I just had to swap it out 😄